src/librustc_resolve/lib.rs

```diff
@@ -1697,9 +1697,10 @@ impl<'a> hir::lowering::Resolver for Resolver<'a> {
         components: &[&str],
         is_value: bool
     ) -> hir::Path {
-        let segments = iter::once(keywords::PathRoot.ident())
+        let segments = crate_root.as_ref().map(|_| keywords::PathRoot.ident()).into_iter()
             .chain(
-                crate_root.into_iter()
+                crate_root.or(Some("crate"))
+                    .into_iter()
```

Comment:
Can't we move this into the callers of `resolve_str_path`? Or I guess we want `None` to mean "this crate"? If the latter, this seems fine to me!

Reply:
Although I'd very much prefer to see `keywords::Crate` rather than `"crate"` that gets interned later.